### PR TITLE
SelectBox FIX: downArrow should not add new items to list (T823045)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

File: js/ui/select_box.js [hunk 1]
```diff
@@ -399,10 +399,11 @@ var SelectBox = DropDownList.inherit({
 
         dataSourceIsLoaded.done((function() {
             var selectedIndex = this._getSelectedIndex(),
+                hasPages = this._dataSource.pageSize(),
```
Comment (DokaRus) [inline js/ui/select_box.js:402]:
```suggestion
                isPageLoad = this._dataSource.paginate(),
```

File: js/ui/select_box.js [hunk 2]
```diff
@@ -399,10 +399,11 @@ var SelectBox = DropDownList.inherit({
...
                 isLastPage = this._dataSource.isLastPage(),
                 isLastItem = selectedIndex === this._items().length - 1;
 
-            if(!isLastPage && isLastItem && step > 0) {
+            if(hasPages && !isLastPage && isLastItem && step > 0) {
```
Comment (DokaRus) [inline js/ui/select_box.js:406]:
```suggestion
            if(isPageLoad && !isLastPage && isLastItem && step > 0) {
```

**Review by DokaRus — COMMENTED** [review]
The data source may have a page size, but the `paginate` option may be disabled

File: js/ui/select_box.js (hunk 1 above)
Comment (ksercs) [reply js/ui/select_box.js:402]: It doesn't work. isLastPage = !paginate(), so this sugg doesn't change the behavior

**Review by DokaRus — APPROVED** [review]
LGTM

**Review by mrjohn0011 — APPROVED** [review]

**Review by San4es — COMMENTED** [review]
LGTM

**Pull request merged**